homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,257 @@
+        translation_key="uptime",
+        key="uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.power_supply if data.state else None,
+        translation_key="power_supply",
+        key="power_supply",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: WatergateConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up all entries for Watergate Platform."""
+
+    coordinator = config_entry.runtime_data
+
+    entities: list[SensorEntity] = [
+        SonicSensor(coordinator, description) for description in DESCRIPTIONS
+    ]
+
+    entities.extend([AutoShutOffEventSensor(coordinator)])
+
+    async_add_entities(entities)
+
+
+class SonicSensor(WatergateEntity, SensorEntity):
+    """Define a Sonic Sensor entity."""
+
+    entity_description: WatergateSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: WatergateDataCoordinator,
+        entity_description: WatergateSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, entity_description.key)
+        self.entity_description = entity_description
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle data update."""
+        self._attr_native_value = self.entity_description.value_fn(
+            self.coordinator.data
+        )
+        self._attr_available = self._attr_native_value is not None
+        self.async_write_ha_state()
+
+    def update(self, value: str | None) -> None:
+        """Update the sensor."""
+        self._attr_native_value = value
+        self.async_write_ha_state()
+
+
+class AutoShutOffEventSensor(WatergateEntity, SensorEntity):
+    """Representation of a sensor showing the latest long flow event."""
+
+    def __init__(
+        self,
+        coordinator: WatergateDataCoordinator,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, "auto_shut_off_event")
+        self.entity_description = SensorEntityDescription(
+            key="auto_shut_off_event",
+            translation_key="auto_shut_off_event",
+            device_class=SensorDeviceClass.TIMESTAMP,
+        )
+
+    def update(self, report: AutoShutOffReport) -> None:
+        """Update the sensor."""
+        self._attr_native_value = datetime.fromtimestamp(report.timestamp, UTC)
+        self._attr_extra_state_attributes = {
+            "type": report.type,
+            "duration": report.duration,
+            "volume": report.volume,
```

Comment:
we don't really like extra_state_attributes (anymore). I'm thinking if it makes sense to put those as their own sensors and model the actual event as `EventEntity`

Reply:
So waht you are saying is that we should in webhook handler trigger an Event (there is no event type for that anyway). But how it is going to be displayed? Because I know that it is not going to be triggered often (like once a year or preferably - never if everything works in your pipework) and I think that those will be visible as an unknown, right?

Reply:
Anyway, I guess that should be in separate PR then if we want to introduce another mechanism (we already introduced translations).

Reply:
ah ok, if that's the case then both an event and its own sensors probably won't make sense. I'm not sure what todo with this then tbh, maybe the others can chime in

Reply:
@CFenner, @epenet any suggestions?

Reply:
I discussed this and this should definitely be a `EventEntity` instead. We can discuss about the `extra_state_attributes` again once you add the event platform.